src/utilities/tleHandling.py

```diff
@@ -0,0 +1,449 @@
+import numpy as np
+import datetime as dt
+import re
+from dataclasses import dataclass, field
+from matplotlib.dates import SEC_PER_DAY
+from Basilisk.utilities import orbitalMotion as om
+
+SPUTNIK_LAUNCHDATE = dt.datetime(1957, 10, 4, 0, 0, 0)
+
+@dataclass
+class TleData:
+    """
+    Data class to hold TLE data (orbit elements) and metadata.
+    """
+    # Required (Orbital elements and time when they are valid is the minimum requirement)
+    oe: om.ClassicElements
+    tleEpoch: dt.datetime
+
+    # Optional
+    satName: str = field(default="BSK-Sat-00")
+    noradID: str = field(default="00000")
+    classification: str = field(default="Unknown")
+    revAtEpoch: float = field(default=0.0)
+    propagator: str = field(default="0")
+    elemSetNo: int = field(default=999)
+    nDot: float = field(default=0.0)
+    nDotDot: float = field(default=0.0)
+    bStar: float = field(default=0.0)
+    launchDate: dt.datetime = field(default=None)
+    launchNo: int = field(default=1)
+    pol: str = field(default="A")
```

Comment:
Nitpick:

Could be nice to add some comments here to the data structure to denote the form of the fields since this is a standardized format.

For example what units are bStar